src/App.js

```diff
@@ -1,24 +1,43 @@
-import logo from './logo.svg';
-import './App.css';
+import React, {useState, useEffect} from 'react';
+import Course from './screens/course/Course';
+import Lesson from './screens/lesson/Lesson';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import { supabase } from './supabaseClient';
 
 function App() {
+  const [courseItems, setCourseItems] = useState([]);
+
+  const getCourseItems = async () => {
+    const { data, error } = await supabase
+      .from('courses')
+      .select('name, id')
+
+    if(error) {
+      throw new Error(error.message)
+    }
+    if(!data) {
+      throw new Error("Course not found")
+    }
+    setCourseItems(data);
+  };
+
+  useEffect(() => {
+    getCourseItems();
+  });
```

Comment:
When this function trigger?

Reply:
useEffect has been called every time that component changed.